content/blog/stackstorm-simple-elegant-automation-for-everyone.md

```diff
@@ -0,0 +1,33 @@
+
+![](https://www.techworldwookie.com/images/actors.png)
+
+Sounds like a lot of code to write. For every system I want to talk to, I will have to write a separate module. For example, in vCenter, I want to build a new port group. Wouldn't it be awesome if the act of creating the port group would automatically trigger something to add that port group information as a new VLAN (virtual LAN) Aruba central. I could write a python script to get that information from vCenter and then refactor it, and use a different API to send the information to Aruba Central. I still wouldn't have the automation I am looking for unless I scheduled a Cron Job to run my code. The task becomes exponentially harder when I want to add another device like HPE OneView into the conversation.
+
+I like writing code, but you need to work smarter, not harder. After all, there is only so much time in the day. That is why I love using python. Python has a lot of pre-written modules that make it extremely extensible. By using the **import** command in my script I can take advantage of modules I have no idea how to even start writing. I like the **plug and play** of how easy it is to consume those extra modules. I like using **requests**, I can just code **import requests**, and I have access to a well developed chunk of code that I did not have to write.
+
+What if we applied the same concept to our problem at the start of this blog? What if each of the resources in the data center had a **module** of sorts, that I could plug into some sort of framework? Something like StackStorm!
+
+![](https://www.techworldwookie.com/images/stackstorm.png)
+
+Before we get too far, here is a quick re-cap of what StackStorm is. StackStorm uses Sensors, Rules, Triggers, Actions, and Workflows. You can think of it as If-this-then-that automation. Based on a event on some device, a StackStorm sensor can detect it, and this could trigger a Rule. The Rule would have some logic in it and if the trigger launched the Rule, the Rule would run an action, or a series of actions called a Workflow. It might sound a little involved but it is all packaged together to make it easy to deploy these solutions.
+
+![](https://www.techworldwookie.com/images/process.png)
+
+What if I told you there was a place called the StackStorm Exchange that already had 170 of these modules just waiting for you to plug into your projects? [Look at this link.](applewebdata://EBBB076B-5028-426E-9291-45124FE61477/exchange.stackstorm.com) It is quite an impressive stockpile of automation just waiting for you. I recently finished developing a StackStorm integration pack for HPE Primera Storage. Before that, I developed a pack to integrate HPE Nimble Storage as well. What communication can we develop? Let's take a look.
```

Comment:
At the end of the 3rd sentence, consider adding "just waiting for you to use." or "just waiting for you to take advantage of it."
In the 5th sentence, take out "as well", placing the period directly after "Nimble Storage."

I was confused by the last two sentences and the first part of the next paragraph until I read further and realized that "these two systems" was referring to the HPE Primera Storage" and the "HPE Nimble Storage". Perhaps a light rephrasing could help:

"Using these  packs, I went about trying to determine what sort of communication could be developed between the two storage systems." (take out "Let's take a look." It's superfluous and complicates things)

Reply:
changed per recommendation